Mathlib/Topology/ContinuousMap/Interval.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2024 Vincent Beffara. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vincent Beffara
+-/
+import Mathlib.Topology.CompactOpen
+import Mathlib.Topology.Order.ProjIcc
+
+/-!
+# Continuous bundled maps on intervals
+
+In this file we prove a few results about `ContinuousMap` when the domain is an interval.
+-/
+
+open Set ContinuousMap Filter Topology
+
+namespace ContinuousMap
+
+variable {α : Type*} [LinearOrder α] [TopologicalSpace α] [OrderTopology α]
+variable {a b c : α} [Fact (a ≤ b)] [Fact (b ≤ c)]
+variable {E : Type*} [TopologicalSpace E]
+
+/-- The embedding into an interval from a sub-interval lying on the left, as a `ContinuousMap`. -/
+def subinterval_left : C(Icc a b, Icc a c) where
+  toFun x := ⟨x, x.2.1, le_trans x.2.2 Fact.out⟩
+
+/-- The embedding into an interval from a sub-interval lying on the right, as a `ContinuousMap`. -/
+def subinterval_right : C(Icc b c, Icc a c) where
+  toFun x := ⟨x, le_trans Fact.out x.2.1, x.2.2⟩
```

Comment:
I think these should be defined in terms of the apparently missing `ContinuousMap.inclusion`

Reply:
I added `ContinuousMap.inclusion`